arangod/Agency/Agent.cpp

```diff
@@ -656,21 +674,15 @@ void Agent::advanceCommitIndex() {
     
   term_t t = _constituent.term();
   {
-    _tiLock.assertNotLockedByCurrentThread();
-    MUTEX_LOCKER(_ioLocker, _ioLock);
-    index_t commitIndex;
-    {
-      CONDITION_LOCKER(guard, _waitForCV);
-      commitIndex = _commitIndex;
-    }
-    if (index > commitIndex) {
+    MUTEX_LOCKER(oLocker, _oLock);
+    if (index > _commitIndex) {
       LOG_TOPIC(TRACE, Logger::AGENCY)
-        << "Critical mass for commiting " << commitIndex + 1
+        << "Critical mass for commiting " << _commitIndex + 1
         << " through " << index << " to read db";
       // Change _readDB and _commitIndex atomically together:
       _readDB.applyLogEntries(
         _state.slices( /* inform others by callbacks */ 
-          commitIndex + 1, index), commitIndex, t, true);
+          _commitIndex + 1, index), _commitIndex, t, true);
 
       CONDITION_LOCKER(guard, _waitForCV);
```

Comment:
wrong lock.  but maybe not needed if previous comment about caching commitLock is wrong.  But, if code continues without caching commitLock then this lock is both wrong and redundant.